tests/v2/actions/observability/stackstate.go

```diff
@@ -0,0 +1,212 @@
+)
+
+// NewStackstateConfiguration is a constructor that takes in the configuration and creates an unstructured type to install the CRD
+func NewStackstateCRDConfiguration(namespace string, stackstateCRDConfig StackStateConfigs) *unstructured.Unstructured {
+
+	crdConfig := &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"metadata": map[string]interface{}{
+				"name":      StackstateName,
+				"namespace": namespace,
+			},
+			"spec": map[string]interface{}{
+				"url":          stackstateCRDConfig.Url,
+				"serviceToken": stackstateCRDConfig.ServiceToken,
+			},
+		},
+	}
+	return crdConfig
+}
+
+// InstallNodeDriver is a helper that utilizes the rancher client and add the stackstate domains to whitelist them.
+func InstallNodeDriver(client *rancher.Client, whitelistDomains []string) error {
+
+	nodedriver := &management.NodeDriver{
+		Name:             StackstateName,
+		Active:           true,
+		WhitelistDomains: whitelistDomains,
+		URL:              "local://",
+		State:            "inactive",
+	}
+
+	stackstateNodeDriver, err := client.Management.NodeDriver.Create(nodedriver)
+	if err != nil {
+		return err
+	}
+
+	err = kwait.PollUntilContextTimeout(context.TODO(), 500*time.Millisecond, defaults.ThirtyMinuteTimeout, true, func(ctx context.Context) (done bool, err error) {
+		resp, err := client.Management.NodeDriver.ByID(stackstateNodeDriver.ID)
+		if err != nil {
+			return false, err
+		}
+
+		if resp.State == "downloading" {
+			return true, nil
+		}
+		return false, nil
+	})
+
+	return err
+}
+
+// InstallStackstateCRD is a helper that utilizes the rancher client and installs the stackstate crds.
+func InstallStackstateCRD(client *rancher.Client) error {
+
+	stackstateCRDConfig := apiextv1.CustomResourceDefinition{
+		TypeMeta:   metav1.TypeMeta{Kind: "CustomResourceDefinition", APIVersion: "apiextensions.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{Name: ObservabilitySteveType},
+		Spec: apiextv1.CustomResourceDefinitionSpec{
+			Group: CrdGroup,
+			Versions: []apiextv1.CustomResourceDefinitionVersion{
+				0: {Name: "v1beta1",
+					Served:  true,
+					Storage: true,
+					Schema: &apiextv1.CustomResourceValidation{
+						OpenAPIV3Schema: &apiextv1.JSONSchemaProps{
+							Type: "object",
+						},
+					},
+				},
+			},
+			Names: apiextv1.CustomResourceDefinitionNames{
+				Plural:   "configurations",
+				Singular: "configuration",
+				Kind:     "Configuration",
+				ListKind: "ConfigurationList",
+			},
+			Scope: "Namespaced",
+		},
+	}
+	crd, err := client.Steve.SteveType(ApiExtenisonsCRD).Create(stackstateCRDConfig)
+	if err != nil {
+		return err
+	}
+
+	client.Session.RegisterCleanupFunc(func() error {
+		err := client.Steve.SteveType(ApiExtenisonsCRD).Delete(crd)
+		if err != nil {
+			return err
+		}
+
+		err = kwait.PollUntilContextTimeout(context.TODO(), 500*time.Millisecond, defaults.ThirtyMinuteTimeout, true, func(ctx context.Context) (done bool, err error) {
```

Comment:
isn't 30 mins too long?

Reply:
in general uninstalling crds take too much time. that was added for other charts for the same reason. but I have reduced it to 2min as this is more of an ui-extension crd than a chart crd.